sklearn/decomposition/_sparse_pca.py

```diff
@@ -294,7 +295,9 @@ def _fit(self, X, n_components, random_state):
             dict_init=dict_init,
             return_n_iter=True,
         )
-        self.components_ = Vt.T
+        # flip eigenvectors' sign to enforce deterministic output
+        code, dictionary = svd_flip(code, dictionary, u_based_decision=False)
```

Comment:
I think we need to include this in "Changed model" since this will give different results for `SparsePCA`.

On a related note, if we are okay with changing the model here, it would also make sense to remove the `sign_flip` parameter in `FastICA`: https://github.com/scikit-learn/scikit-learn/blob/f0af898de5bfa93916b659438b4cf0b86fffd974/sklearn/decomposition/_fastica.py#L264

(The `flip_sign` parameter was not released yet)

Reply:
I'm +1 to remove the param in FastICA and always flip sign before it's released